packages/runtime/src/runtimes/react/index.tsx

```diff
@@ -122,8 +123,10 @@ export function RuntimeProvider({
     async function setReactBuilderPreviewStore(): Promise<void> {
       const ReactBuilderPreview = await import('../../state/react-builder-preview')
 
-      setStore(store =>
-        ReactBuilderPreview.configureStore({ preloadedState: store.getState(), client }),
+      startTransition(() =>
+        setStore(store =>
+          ReactBuilderPreview.configureStore({ preloadedState: store.getState(), client }),
+        ),
```

Comment:
If this causes a large delay in the builder we might want to consider other approaches to using the builder store

Reply:
One approach would be to lazily load different providers. Something like this:

```tsx
const PreviewProvider = dynamic(() => import('./path/to/preview-provider'))
const LiveProvider = dynamic(() => import('./path/to/live-provider'))

export function Provider({ isPreview }) {
  return isPreview ? <PreviewProvider /> : <LiveProvider />
}
```

Reply:
Updated.

But currently having an error on the initial render in the builder for this commit:
<img width="570" alt="image" src="https://user-images.githubusercontent.com/13066728/211371235-1cea4c2c-eb6d-48a6-9074-375f001b2adc.png">

Will work on this again tomorrow.

Reply:
That looks like initialization isn't being done properly—specifically the effect that registers the document with the store.

Reply:
Cool. I got it fixed!